### DISCUSSION TITLE
JSON Web Token (JWT) Embedded Tokens

### DISCUSSION THREAD

**Discussion by schrieveslaach** [keycloak/keycloak#28550] [Q&A]
There is a [Internet-Draft that describes a way of embedding tokens by value](https://datatracker.ietf.org/doc/draft-yusef-oauth-nested-jwt/) and I was wondering if this can be implemented with a custom protocol mapper. Any advice or even code snippets.

**Comment by justin-tay** [comment]
Given that this uses a special grant type `urn:ietf:params:oauth:grant-type:embedded-tokens`, I think you will need to use the new OAuth Grant Type SPI introduced in Keycloak 24 to do this.

- https://github.com/keycloak/keycloak/pull/26251

You can look at the existing implementations for how you might go about implementing it https://github.com/keycloak/keycloak/pull/26251/files
